### PR TITLE
New samples: Kueue cohort tutorial

### PR DESCRIPTION
**Pull request by rbarberop**
This PR contains the yaml examples for the new Kueue tutorial using cohorts for resource sharing.

Code is v1beta1

### CODE REVIEW

**Comment by rbarberop** [comment]
@alculquicondor FYI

**Comment by rbarberop** [comment]
@Shabirmean https://github.com/GoogleCloudPlatform/kubernetes-engine-samples/pull/498 closed in favor of this one

File: batch/kueue-cohort/team-a-job.yaml
Comment (alculquicondor) [inline batch/kueue-cohort/team-a-job.yaml:27]:
```suggestion
  suspend: true # Set to true and allow Kueue to control when the Job starts
```

File: batch/kueue-cohort/team-b-job.yaml
Comment (alculquicondor) [inline batch/kueue-cohort/team-b-job.yaml:27]: same comment

**Review by alculquicondor — COMMENTED** [review]
/lgtm

**Review by bourgeoisor — APPROVED** [review]
LGTM, let me know when this is good to merge.

File: batch/kueue-cohort/team-a-cq.yaml
Comment (alculquicondor) [inline batch/kueue-cohort/team-a-cq.yaml:22]: this namespaceSelector should match the team-a namespace.

This way, users from team-b can't directly submit jobs here (they can still borrow, through team-b-cq)

Comment (alculquicondor) [reply batch/kueue-cohort/team-a-cq.yaml:22]:
```yaml
namespaceSelector:
  matchLabels:
    kubernetes.io/metadata.name: team-a
```

File: batch/kueue-cohort/spot-cq.yaml
Comment (alculquicondor) [inline batch/kueue-cohort/spot-cq.yaml:22]: you could remove this line: Then noboby is able to send jobs directly to this clusterqueue, but clusterqueues in the cohort can borrow.

File: batch/kueue-cohort/flavors.yaml
Comment (alculquicondor) [inline batch/kueue-cohort/flavors.yaml:30]:
```yaml
cloud.google.com/gke-provisioning: spot
```

File: batch/kueue-cohort/flavors.yaml
Comment (alculquicondor) [inline batch/kueue-cohort/flavors.yaml:22]:
```yaml
cloud.google.com/gke-provisioning: standard
```

File: batch/kueue-cohort/team-b-job.yaml
Comment (alculquicondor) [inline batch/kueue-cohort/team-b-job.yaml:22]:
```suggestion
    kueue.x-k8s.io/queue-name: team-b-lq # Point to the LocalQueue
```

File: batch/kueue-cohort/create_jobs.sh
Comment (alculquicondor) [inline batch/kueue-cohort/create_jobs.sh:21]: did you intend to have all these lines? You should be able to control the rate by changing the sleep time

**Review by alculquicondor — COMMENTED** [review]
LGTM

**Comment by rbarberop** [comment]
Blocked on https://github.com/kubernetes-sigs/kueue/issues/659

**Pull request merged**
